panel/panel.py

```diff
@@ -136,64 +119,51 @@ def run_update_panel_cmd(panel, text=None):
     panel.run_command(cmd, {'text': text, 'clear_sel': clear_sel})
 
 
-def format_row(lineno, error_type, dic):
-    lineno = int(lineno) + 1
-    start = dic['start'] + 1
-    msg = dic['msg'].rstrip()
-    tmpl = " {LINENO:>5}:{START:<4} {ERR_TYPE:7} {linter:>12}: {code:12} {MSG}"
-    return tmpl.format(
-        LINENO=lineno, START=start, ERR_TYPE=error_type, MSG=msg, **dic)
+def format_row(item):
+    line = int(item["line"]) + 1
+    start = item['start'] + 1
+    tmpl = " {LINE:>5}:{START:<4} {error_type:7} {linter:>12}: {code:12} {msg}"
+    return tmpl.format(LINE=line, START=start, **item)
 
 
-def filter_ok(check_val, key, panel_filter):
+def passes_listing(buf_d, panel_filter):
     """Check value against white- and blacklisting and return bool."""
     # check whitelisting
     if not panel_filter:
         return True
 
-    if key in panel_filter and check_val not in panel_filter[key]:
-        return False
+    x = {"types": "error_type", "codes": "code", "linter": "linter"}
```

Comment:
what is this a map from ? error keys to style setting keys ?

Reply:
It is for command args you use for white-/ blacklisting to keys in `raw_errors`.

Reply:
COMMAND_ARG_TO_ERROR_KEY

Reply:
Upper case, bc it is a const? I have seldom seen const been declared in function. Usually they are in global scope at top at the file.

Reply:
But it *is* a constant mapping regardless where you define it